cloudfoundry/resource_cf_app.go

```diff
@@ -746,14 +746,6 @@ func resourceAppDelete(ctx context.Context, d *schema.ResourceData, meta interfa
 		return diag.FromErr(err)
 	}
 
-	jobURL, _, err := session.ClientV3.DeleteApplication(appGUID)
-	if err != nil && !IsErrNotFound(err) {
-		return diag.FromErr(err)
-	}
-
-	err = PollAsyncJob(PollingConfig{
-		Session: session,
-		JobURL:  jobURL,
-	})
+	err = v3appdeployers.SafeAppDeletion(*session.ClientV3, appGUID, 5)
```

Comment:
Small note: this would mean that we can potentially be waiting up to 5 minutes for each venerable app deletion. We'd probably want to discuss with @loafoe on whether this is too much time or not.

Reply:
Hm, you are right. I'm not sure what actually will take up time in the process. The assumption is that if deletion of services fails (due to asynchronicity), it does fail fast.

See the delete action at cloud controller:
- first, they delete the nontransactional resources (the binding): https://github.com/cloudfoundry/cloud_controller_ng/blob/ee25971edd577137ecc1e7104ad343e55043de9f/app/actions/app_delete.rb#L41
- only afterwards, they delete packages, tasks, droplets, revisions, ..... https://github.com/cloudfoundry/cloud_controller_ng/blob/ee25971edd577137ecc1e7104ad343e55043de9f/app/actions/app_delete.rb#L46

As such, when the retry happens, it will happen fast and not after lots of iterations in the polling loop.